### PR TITLE
Reduce nginx basic auth for whitehall-admin

### PR DESCRIPTION
**Pull request by alexmuller**
## Todo

- [x] Get sign-off
- [ ] Verify in integration

### CODE REVIEW

**Comment by elliotcm** [comment]
We've got to keep this on fact check requests so

**Comment by alexmuller** [comment]
Updated this to reduce the amount of basic auth rather than removing it entirely. Needs a bit of careful testing in integration.

**Comment by elliotcm** [comment]
Awesome.  Needs a rebase.

Should we be deploying this branch to integration then?  Presumably merging it could block production deploys otherwise.

**Comment by alexmuller** [comment]
I've just tested this locally and I'm happy with the way it behaves. If somebody thinks this looks ok I think we should be alright to merge this.

File: modules/govuk/manifests/app/nginx_vhost.pp [hunk 1]
```diff
@@ -4,6 +4,7 @@
   $app_port,
   $aliases = [],
   $protected = false,
+  $protected_location = '/',
```
Comment (rjw1) [inline modules/govuk/manifests/app/nginx_vhost.pp]: puppetdocs?

File: modules/nginx/manifests/config/vhost/proxy.pp [hunk 2]
```diff
@@ -29,6 +29,12 @@
 # [*alert_5xx_critical_rate*]
 #   The error percentage that triggers a critical alert
 #
+# [*protected*]
+#   Boolean, whether or not the vhost should be protected with basic auth
+#
+# [*protected_location*]
+#   Prefix path to protect with basic auth, defaults to everything
+#
 # TODO: More docs!
```
Comment (rjw1) [inline modules/nginx/manifests/config/vhost/proxy.pp]: since your editing this file how about doing the todo?

Comment (alexmuller) [inline modules/nginx/manifests/config/vhost/proxy.pp]: You're killing me bob, I have added all the docs.

**Comment by rjw1** [comment]
:+1:

**Pull request merged**